ext/DDProf/src/MurmurHash3.cpp

```diff
@@ -0,0 +1,348 @@
+//-----------------------------------------------------------------------------
```

Comment:
What's the difference between this MurmurHash3 and the C version in `ext/MurmurHash3`? Are we not able to use just one implementation?

Reply:
I don't think I meant to commit both of them; will look into that. Thanks!

Reply:
I extracted the hashing to `ext/DatadogStringHash`, which exposes C99 and C++14 APIs:

- `datadog_string_hash` for C99+
- `datadog::string_hash` for C++14+

For now only the C++ API is used in this PR.

The API still uses MurmurHash3, but I made it so none of the symbols are exposed. This way we can shared the string hashing bits across profiling and tracing if we need to, but we won't have linker issues with other libraries.